pkgs/by-name/gh/gh-copilot/update.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl jq common-updater-scripts nix-prefetch
+
+set -euo pipefail
+set -x
+
+ROOT="$(dirname "$(readlink -f "$0")")"
+NIX_DRV="$ROOT/package.nix"
+if [ ! -f "$NIX_DRV" ]; then
+  echo "ERROR: cannot find gh-copilot in $ROOT"
+  exit 1
+fi
+
+fetch_arch() {
+  VER="$1"; ARCH="$2"
+  URL="https://github.com/github/gh-copilot/releases/download/v${VER}/${ARCH}";
+  nix-prefetch "{ stdenv, fetchzip }:
+stdenv.mkDerivation rec {
+  pname = \"vere\"; version = \"${VER}\";
+  src = fetchurl { url = \"$URL\"; };
+}
+"
+}
+
+replace_sha() {
+  # https://stackoverflow.com/a/38470458/22235705
+  sed -rziE "s@($1[^\n]*\n[^\n]*hash = )\"sha256-.{44}\";@\1\"$2\";@" "$NIX_DRV"
```

Comment:
Sorry for this monstrosity

Reply:
To make this more reviewable (so it would hopefully merge), here is an explanation on what does this do

- `-r`, `-E` - --regexp-extended
- `-z` - separate lines by NUL characters
- `-i` - edit files in place

---

- `$1` looks for architecture on the previous line ($1 is the first argument)
- `[^\n]*\n[^\n]*` is from stackoverflow answer linked in the comment, it just selects everything until next line
- `hash = )\"` - on the next line, we look for a hash, `)` closes capture group with architecture line
- `sha256-.{44}` - hash starts with `sha256-` and 44 random characters (/44 dots in regex)
- `\";@` then it is the end of the line, and we proceed to replace part
- `\1` - capture group one, this is architecture line plus everything until `"sha256`
- `\"$2\"` - second argument, which is hash from nix-prefetch
- `;@` end of the sed expression

Reply:
What is programming without a few monstrosities

Reply:
Instead of doing such a scary script that abuses of GNU extensions, why not the good old awk?

Reply:
How would it look like using awk? I'm sorry, I'm not really proficient in awk and generally in bash

Reply:
Well, if the idea of your script is to pick the previous line when the current line satisfies some condition  (namely, having the string "hash"), then you need a placeholder variable that saves that previous line.

Or, thinking in reverse:
- save the current line (`$0` in Awk lingo) in a variable, say `previousLine = $0`;
- Loop:
  - If the current line matches a regex, then you have the previous line to work
  - else, get back to loop

I will try to come with some code later.